components/script/dom/window.rs

```diff
@@ -1445,6 +1451,12 @@ impl Window {
         None
     }
 
+    pub fn freeze(&self) {
+        self.upcast::<GlobalScope>().suspend();
+        // A hint to the JS runtime that now would be a good time to GC this window.
+        self.Gc();
```

Comment:
We are GCing user script data and unattached DOM nodes here. Attached DOM nodes can't be reclaimed.